Parse/PFFileUploadController.h

```diff
@@ -0,0 +1,34 @@
+//
+//  PFUploadController.h
+//  Parse
+//
+//  Created by Ken Cooper on 2/20/17.
+//  Copyright © 2017 Parse Inc. All rights reserved.
+//
+
+#import <Foundation/Foundation.h>
+#import <Bolts/BFTask.h>
+#import "PFFileUploadResult.h"
+
+/**
+ A policy interface for overriding the default upload behavior of uploading a PFFile
+ to application's parse server. Allows for direct uploads to other file storage
+ providers.
+ */
+@protocol PFFileUploadController <NSObject>
```

Comment:
we could very well have the `PFRESTFileUploadController: NSObject<PFFileUploadController>` what do you think?